### ISSUE TITLE
Accound deactivation warning is not clear enough

### ISSUE THREAD

**Issue by pafcu** [vector-im/riot-web#5148]
### Description

It seems that the warning shown when a user deactivates their account is not clear enough. Specifically, it seems that it is not always clear that the user id can not be registered. Possible reasons include confusing terminology (what is a user ID?), difficulty understanding the language (if non-native), lack of attention, etc.

Perhaps some sort of generally understood warning icon could be added to draw attention to the fact that something potentially bad is about to happen.

Also, like in some system, perhaps require the user to type in their user id, as that may strengthen the point that what they are about to do is really, really important.

Finally, the key part of the text could perhaps be in bold.

**Comment by pafcu** [comment]
It would probably also be good to actually mention which account is being deactivated

**Comment by pafcu** [comment]
"Deactive" might also be problematic terminology since in e.g. Facebook deactivation means that you _can_ reactivate your account (as opposed to deleting it).

**Comment by lampholder** [comment]
I agree with all of the above - do you have any concrete/anecdotal evidence of this tripping people up?

**Comment by pafcu** [comment]
You get people in Matrix HQ and Riot asking about it once in a while. Not super often, like a few times per months maybe? But factoring in that

1. Account deactivation is not that common to begin with
2. Not everyone who missed it actually end up trying to reactivate
3. Not everyone who missed it, and tried to reactivate ask how to fix it online

it may or may not be a common-ish problem.

**Comment by aaronraimist** [comment]
Some other services have specific checkboxes that you have to click before they let you delete an account

You could add something like
- [ ] You understand you won't be able to re-register the same user ID

**Comment by harmancasvi** [comment]
Hello I just had a problem with this. I've been using element with the matrix server for a while, I'm a sysadmin as well so I know what I'm doing (I like to believe that's the case at least).

The thing is, I just installed element on my desktop, there's always some things on the settings I like to tweak as you may  think. I was going through the settings and came across the Deactivate function. I have previously been there but it was a long time ago when trying to learn the app features etc. So I wanted to go check what has changed in the mean time with the desktop client. BUT NOT DEACTIVATE MY FREACKING ACCOUNT IN A SINGLE CLIC WTF!?

Long story short I pressed the button that said "Deactivate my account" waiting for some confirmation or something, and then go back, just wanted to check the warning message and the confirmation method used by the client app. That's exactly what I was trying to check, as I've done it before on mobile but well oh well... While I was starting to read the warning message (yeah, there was one) it all went south, in the mid of reading the dialog I was kicked out from both the desktop app and the mobile client. Then I tried to login back and got the "This account has been deactivated message." 🤦‍♂️ 

WHAT THE HECK MAN?? Just like that I lost my account without confirmation or anything? Is this supposed to be like that or is it just a VERY STUPID and dangerous bug in the app?

For the record, when I was reading the app I didn't touch anything on the mouse, keyboard or anything, as I wasn't trying to actually deactivate my account, I was just checking the app and the new features/UI if any.

The realization of what happened was immediate, as I had my mobile app open because I just verified the desktop app with the mobile one. I couldn't believe that it deactivated my account without notice or any sort of confirmation.

And to think I was 🤏  this close to deploy an instance and use element on my work place. Never again will think about it with such a dangerous oversee 😞

**Comment by turt2live** [comment]
@harmancasvi the bug you're seeing is something else entirely. It'd be good to know what your server setup is in a new issue as I suspect it's that and not Element which caused the deactivation to happen.

**Comment by harmancasvi** [comment]
@turt2live it's the main matrix public server, I wasn't hosting my own. If that's the reference server it leaves a lot to be desired if the app was not the culprit IMO.

**Comment by turt2live** [comment]
please open a new issue so we can look into it.

**Comment by harmancasvi** [comment]
@turt2live I'm going o a trip now and have no more time to explain this, as I said I no longer care about this service, so I will not open an issue just to let it there and not provide useful info to follow up. I just wanted to make sure you guys knew what happened to me so you can look into it.

If someone replies on issue 787 so be it, else I at leas let you know.

Thank you for the info anyway.

**Comment by BrenBarn** [comment]
This issue still isn't resolved.  "Deactivate" is a poor choice of terminology, since deactivation implies activation, which isn't actually possible.  It would be better to call it "delete".

**Comment by aaronraimist** [comment]
Delete would be a nice word to use but it implies the account would actually be deleted, which is untrue. The account still exists in the database after deactivation. The messages you sent are still there and no one can ever use that username again. To me delete would imply the account is completely gone.

Deactivate is the word that most other services seem to use for the same concept. Twitter, Slack, and Facebook call it deactivate. Discord calls it disable, would that be more acceptable?

**Comment by BrenBarn** [comment]
As far as I can tell, those other services do not use "deactivate" for the same concept.  See [this Facebook page](https://www.facebook.com/help/212666185422169), [this Twitter page](https://help.twitter.com/en/managing-your-account/trouble-reactivating-twitter-account), and [this Slack page](https://slack.com/help/articles/360055665434-Reactivate-your-Slack-account).  All describe a mechanism for reactivating accounts.  In the case of Twitter this can only be done for a limited time; for Slack it can only be done by a workspace admin.  But those are not like Matrix, where "deactivate" means "immediately and permanently lock me out of this".

I understand that the account is not "fully" deleted, but what I'm saying is that by far the most important thing to convey to the user is "you will never be able to use this account again", and I think "delete" conveys that much better than "deactivate".  It is much better to say "delete" and use some confirmation dialog that people may not read to mention caveats about what won't be deleted than to say "deactivate" and then have people not read the dialog that explains that it is actually permanent.

I suppose an alternative would be to change every occurrence of "deactivate account" everywhere in Element to "permanently deactivate account" to really hammer home that point.

**Comment by aaronraimist** [comment]
> for Slack it can only be done by a workspace admin

That's the same for Matrix. Your server admin can reactivate your account but you cannot.

But yes I take your point that Facebook and others let you change your mind. Changing it to "Permanently deactivate account" makes sense to me.
